src/main.ts

```diff
@@ -88,15 +88,16 @@ export function validateObjectTree(iobObjects: ObjectWithValue[]): void {
 	// Foreach sub-state (. in id)
 	iobObjects.filter(item => item.object.type.split(".").length > 0).filter(item => item.object.type === "state").forEach(iobObj => {
 		const iobIDPath = iobObj.id.split(".");
-		if (iobIDPath.length > 0){
+		if (iobIDPath.length > 2){
 			// Remove last element (=name of state)
 			iobIDPath.pop();
 			let iobIDBasePath = "";
+			let CountPathDepth = 0;
 			iobIDPath.forEach(iobIDName =>{
 				if (iobObjects.filter(item => item.id === `${iobIDBasePath}${iobIDName}`).length > 1){
 					throw `Duplicated object ${iobIDBasePath}${iobIDName} defined`
 				}else{
-					if (iobObjects.filter(item => item.id === `${iobIDBasePath}${iobIDName}`).length === 0){
+					if (CountPathDepth >= 2 && iobObjects.filter(item => item.id === `${iobIDBasePath}${iobIDName}`).length === 0){
```

Comment:
`array.filter(...).length === 0` can be written more succinct as `!array.some(...)` (and is probably more performant because it will return false as soon as one item satisfies the check.

And what is the logic behind checking if `iobIDPath.length > 2` and `CountPathDepth >= 2`. I don't really understand this 🤷🏻‍♂️

Reply:
```array.filter(...).length === 0``` --> right, I will fix this.

At the begining the id of a state was just "state", now it is "adapter.instance.state". Foreign objects can only exist at the third level, not for "adapter" or "instance". So I've integrated a counter to determine the "path depth" of the current ID. When the path depth >= 2 then we are behind "adapter.instance".

Reply:
gotcha